### PR TITLE
feat: Implement `hasValue` OUT parameter for iOS

### PR DESCRIPTION
**Pull request by mrousavy**
Follow-up PR of #828 to implement the OUT parameter for iOS as well.

I specifically did not implement the `hasValue` OUT parameter for Android (yet), because
1. I do not need it for Android myself
2. On Android you cannot pass pointers/booleans by reference. We would have to introduce a new class which holds a boolean reference, and I am not sure if this is something that we want to support? cc @lingol

### CODE REVIEW

**Comment by mrousavy** [comment]
On Android, we need something like this:

```java
public class Ref<T> {
    private T value;

    public Ref(T value) {
        this.value = value;
    }

    public void set(T value) {
        this.value = value;
    }
    public T get() {
        return this.value;
    }
}
```

Then pass that as a "out" parameter:

```java
public class MMKV ... {
    ...

    public boolean getBoolean(String key, boolean defValue, Ref<boolean> out_hasValue) {
        return decodeBool(nativeHandle, key, defValue, out_hasValue);
    }

    ...
```

but this looks a bit ugly in my opinion. Instead, a simple solution would work great here:

```java
public class MMKV ... {
    ...

    public Boolean getBoolean(String key, boolean defValue) {
        return decodeBool(nativeHandle, key, defValue);
    }

    ...
```

Since we now return a boxed boolean here (`Boolean` instead of `boolean`) we can also return `null` :)
But I think you said we don't want any breaking changes here

File: iOS/MMKV/MMKV/libMMKV.mm [hunk 1]
```diff
@@ -386,7 +386,10 @@ - (BOOL)getBoolForKey:(NSString *)key {
     return [self getBoolForKey:key defaultValue:FALSE];
```
[the remark this answers is not in the document]
Comment (mrousavy) [reply iOS/MMKV/MMKV/libMMKV.mm]: fixed

**Comment by lingol** [comment]
> On Android, we need something like this:
> 
> ```java
> public class Ref<T> {
>     private T value;
> 
>     public Ref(T value) {
>         this.value = value;
>     }
> 
>     public void set(T value) {
>         this.value = value;
>     }
>     public T get() {
>         return this.value;
>     }
> }
> ```
> 
> Then pass that as a "out" parameter:
> 
> ```java
> public class MMKV ... {
>     ...
> 
>     public boolean getBoolean(String key, boolean defValue, Ref<boolean> out_hasValue) {
>         return decodeBool(nativeHandle, key, defValue, out_hasValue);
>     }
> 
>     ...
> ```
> 
> but this looks a bit ugly in my opinion. Instead, a simple solution would work great here:
> 
> ```java
> public class MMKV ... {
>     ...
> 
>     public Boolean getBoolean(String key, boolean defValue) {
>         return decodeBool(nativeHandle, key, defValue);
>     }
> 
>     ...
> ```
> 
> Since we now return a boxed boolean here (`Boolean` instead of `boolean`) we can also return `null` :) But I think you said we don't want any breaking changes here

You're right.  Returning a boxed value is absolutely a NO.

**Comment by mrousavy** [comment]
Okay got it. Then we'll leave out the feature for Android for now, since you do not want to introduce breaking changes, right?

Also, do you have a timeline for the next release?

**Comment by lingol** [comment]
No timeline for now. Is your RN lib in a timeline or something like that?

**Comment by mrousavy** [comment]
No timeline, I'm just preparing to use the new `hasValue` OUT parameter in my PR https://github.com/mrousavy/react-native-mmkv/pull/306 and ship an update once that's available

**Comment by lingol** [comment]
Okay. I will try to make it happen this month.

**Pull request merged**